addons/spreadsheet_dashboard/models/spreadsheet_dashboard_group.py

```diff
@@ -19,3 +19,28 @@ def _unlink_except_spreadsheet_data(self):
             external_id = external_ids[group.id]
             if external_id and not external_id.startswith('__export__'):
                 raise UserError(_("You cannot delete %s as it is used in another module.", group.name))
+
+    @api.model
+    def get_dashboard_groups_and_favorites(self):
```

Comment:
do we really need this new public method ?
couldn't we add `is_user_favorite` to the web search read spec ? (see the other comment) And derive the favorited ones client-side.
The new method is a bit cumbersome (needs to be tested, needs to be mocked in client-side tests)